corehq/warehouse/models.py

```diff
@@ -62,3 +62,88 @@ def fail(self):
     )
     def decommission(self):
         return
+
+
+class BaseDim(models.Model):
+    domain = models.CharField(max_length=255)
+
+    dim_last_modified = models.DateTimeField(auto_now=True)
+    dim_created_on = models.DateTimeField(auto_now_add=True)
+
+    class Meta:
+        abstract = True
+
+
+class UserDim(BaseDim):
+    user_id = models.CharField(max_length=255)
+    username = models.CharField(max_length=255)
+    first_name = models.CharField(max_length=255)
+    last_name = models.CharField(max_length=255)
+    email = models.CharField(max_length=255)
+    user_type = models.CharField(max_length=255)
+
+    is_active = models.BooleanField()
+    is_staff = models.BooleanField()
+    is_superuser = models.BooleanField()
+
+    last_login = models.DateTimeField()
+    date_joined = models.DateTimeField()
+
+
+class GroupDim(BaseDim):
+    group_id = models.CharField(max_length=255)
+    name = models.CharField(max_length=255)
+
+    case_sharing = models.BooleanField()
+    reporting = models.BooleanField()
+
+    group_last_modified = models.DateTimeField()
+
+
+class LocationDim(BaseDim):
```

Comment:
my biggest question for modeling the location dimension is how we should model the hierarchy. in the kimball guide, it talks about using a "Slightly ragged" hierarchy for models that have a short depth and are variable. it'd be great to talk through whether you think this is a good approach or not. my other concern is that we actually don't place any limits on how deep the location hierarchy can go. perhaps we can start enforcing that?